src/Cryptol/Eval.hs

```diff
@@ -460,10 +465,9 @@ etaDelay sym msg env0 Forall{ sVars = vs0, sType = tp0 } = goTpVars env0 vs0
       TVRec fs ->
           do v' <- sDelay sym (Just msg) (fromVRecord <$> v)
              let err f = evalPanic "expected record value with field" [show f]
-             return $ VRecord $ Map.fromList
-                [ (f, go t =<< (fromMaybe (err f) . Map.lookup f <$> v'))
-                | (f,t) <- fs
-                ]
+             let eta f t = Identity (go t =<< (fromMaybe (err f) . lookupField f <$> v'))
+             let fs' = runIdentity (traverseRecordMap eta fs)
```

Comment:
This would be a good place to use a `mapWithKey`-style function (see comment on `RecordMap.hs`)